src/core/layout/qgslayoutitempolyline.cpp

```diff
@@ -312,6 +313,19 @@ void QgsLayoutItemPolyline::setArrowHeadWidth( double width )
   update();
 }
 
+QPainterPath QgsLayoutItemPolyline::shape() const
+{
+  QPainterPath path;
+  path.addPolygon( mPolygon );
+
+  QPainterPathStroker ps;
+
+  ps.setWidth( mPolylineStyleSymbol->width() );
```

Comment:
One issue here is that QgsLineSymbol::width() doesn't correctly handle unit types. At the moment this is assuming that all line widths match the same units as the layout (e.g. millimeters), but the sizing will be wrong whenever either the layout or the line symbol aren't in millimeters. 

Are you able to reuse mMaxSymbolBleed here?